game/core/game_view.py

```diff
@@ -129,21 +133,32 @@ def on_mouse_press(
     @beartype
     def on_key_press(self, key: int, modifiers: int) -> None:
         """React to key press."""
-        self.pressed_keys.pressed(key, modifiers)
-        self.rpg_movement.on_key_press(key, modifiers)
-        # Convenience handlers for Reload and Quit
-        meta_keys = {arcade.key.MOD_COMMAND, arcade.key.MOD_CTRL}
-        if key == arcade.key.R and modifiers in meta_keys:
-            logger.warning("Reloading modules")
-            self.reload_modules()
-        if key == arcade.key.Q and modifiers in meta_keys:  # pragma: no cover
-            logger.error("Received Keyboard Shortcut to Quit")
-            arcade.exit()  # type: ignore[no-untyped-call]
-        if key == arcade.key.ESCAPE:
-            self.window.show_view(PauseMenu(self))  # type: ignore[has-type]
-        for register in self.get_all_registers():
-            if register.on_key_press:
-                register.on_key_press(key, modifiers)
+        try:
+            if not self.disable_movement:
+                self.pressed_keys.pressed(key, modifiers)
+                self.rpg_movement.on_key_press(key, modifiers)
+            else:
+                self.gui.draw_message_box(
+                    message=f"Before you can move, you must select your character!",
+                    notes="Edit the code in 'task01/task_s_select_character.py' to select your character",
+                    seconds=5,
+                )
```

Comment:
Nit, but I always prefer to have the True statement first:

```suggestion
            if self.disable_movement:
                self.gui.draw_message_box(
                    message=f"Before you can move, you must select your character!",
                    notes="Edit the code in 'task01/task_s_select_character.py' to select your character",
                    seconds=5,
                )
            else:
                self.pressed_keys.pressed(key, modifiers)
                self.rpg_movement.on_key_press(key, modifiers)
```

Reply:
Ah gotcha, I usually prefer having the "expected" behavior first, but I'll defer to you